internal/gcsx/bucket_manager_test.go

```diff
@@ -120,3 +119,33 @@ func (t *BucketManagerTest) TestSetUpBucketMethod() {
 	ExpectNe(nil, bucket.Syncer)
 	ExpectEq(nil, err)
 }
+
+func (t *BucketManagerTest) TestSetUpBucketMethodWhenBucketDoesNotExist() {
+	var bm bucketManager
+	bucketConfig := BucketConfig{
+		BillingProject:                     "BillingProject",
+		OnlyDir:                            "OnlyDir",
+		EgressBandwidthLimitBytesPerSecond: 7,
+		OpRateLimitHz:                      11,
+		StatCacheCapacity:                  100,
+		StatCacheTTL:                       20 * time.Second,
+		EnableMonitoring:                   true,
+		DebugGCS:                           true,
+		AppendThreshold:                    2,
+		TmpObjectPrefix:                    "TmpObjectPrefix",
+		EnableStorageClientLibrary:         true,
+	}
+	ctx := context.Background()
+	bm.storageHandle = t.storageHandle
+	bm.config = bucketConfig
+	bm.gcCtx = ctx
+	bm.conn = &Connection{
+		wrapped: gcsfake.NewConn(timeutil.RealClock()),
+	}
+
+	bucket, err := bm.SetUpBucket(context.Background(), invalidBucketName)
+
+	// Error in iterating through objects: storage: bucket doesn't exist
+	ExpectNe(nil, err)
```

Comment:
are we not going to validate the error message?

Reply:
It's because it is a string, and if [this](https://github.com/GoogleCloudPlatform/gcsfuse/blob/master/internal/storage/bucket_handle.go#L234) message changes in the future, the test can fail; that's why I didn't compare the string error message and just checked if it is non-nil.

Reply:
fixed